tsdb/index/tsi1/tag_block.go

```diff
@@ -363,6 +367,34 @@ func (e *TagBlockValueElem) SeriesIDs() ([]uint64, error) {
 	return a, nil
 }
 
+// SeriesIDSet returns a set of series ids.
+func (e *TagBlockValueElem) SeriesIDSet() (*tsdb.SeriesIDSet, error) {
+	ss := tsdb.NewSeriesIDSet()
+
+	// Read bitmap data directly from mmap, if available.
+	if e.seriesIDSetData != nil {
+		if err := ss.UnmarshalBinaryUnsafe(e.seriesIDSetData); err != nil {
+			return nil, err
+		}
+		return ss, nil
+	}
+
+	// Otherwise decode series ids from uvarint encoding.
```

Comment:
Can't we memoize this? Otherwise we're going to be building a series id set each time the block is read until the next TSI compaction kicks in and converts to the bitmap.

Reply:
It doesn't seem like it's worth trying to optimize index files that are just going to get recompacted eventually. It also could add a lot to the heap if there are a lot of tag key values.